### PR TITLE
drivers/tcs37727 : Expose Configurations to Kconfig

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in TCS37727 Sensor Device driver to Kconfig.

### Testing procedure
1. New documentation was built using doxygen 

The build worked fine.

2. New macro was introduced in tests/driver_tcs37727/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-246-g68166-Kconfig_tcs37727_tests)
CONFIG_TCS37727_ATIME_DEFAULT=200000
TCS37727 RGBC Data; Sensor driver test application

Initializing first configured TCS37727 sensor...[tcs37727] init: error while reading ID register
[Failed]

#### Usage with CFLAGS 

/tests/driver_tcs37727/Makefile

> CFLAGS += -DCONFIG_TCS37727_ATIME_DEFAULT=15000

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-246-g68166-Kconfig_tcs37727_tests)
CONFIG_TCS37727_ATIME_DEFAULT=15000
TCS37727 RGBC Data; Sensor driver test application

Initializing first configured TCS37727 sensor...[tcs37727] init: error while reading ID register
[Failed]

#### Usage with Kconfig

/tests/driver_tcs37727/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-246-g68166-Kconfig_tcs37727_tests)
CONFIG_TCS37727_ATIME_DEFAULT=250000
TCS37727 RGBC Data; Sensor driver test application

Initializing first configured TCS37727 sensor...[tcs37727] init: error while reading ID register
[Failed]

Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri

### CODE REVIEW

File: drivers/include/tcs37727.h [hunk 1]
```diff
@@ -47,10 +47,12 @@ extern "C"
  * @{
  */
 /**
- * @brief   Default RGBC integration time.
+ * @brief   Default RGBC integration time in microseconds.
+ *
+ * Refer section "RGBC Time Register" in datasheet for more information.
```
Comment (leandrolanzieri) [inline drivers/include/tcs37727.h:52]:
```suggestion
 * Refer to the section "RGBC Time Register" in the datasheet for more information.
```

File: drivers/tcs37727/Kconfig [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+if KCONFIG_MODULE_TCS37727
+
+config TCS37727_ATIME_DEFAULT
+    int "RGBC integration time in microseconds"
+    range 2400 614000
+    default 200000
+    help
+        RGBC integration time impacts both the resolution and
+        the sensitivity of the RGBC reading.
+        Refer section "RGBC Time Register" in datasheet for more information.
```
Comment (leandrolanzieri) [inline drivers/tcs37727/Kconfig:22]:
```suggestion
        Refer to the section "RGBC Time Register" in the datasheet for more information.
```

**Comment by leandrolanzieri** [comment]
Please rebase

**Comment by akshaim** [comment]
> Please rebase

Done. Thanks for the review.

**Pull request merged**